### PR TITLE
prow: protect experimental branch

### PR DESCRIPTION
**Pull request by zirain**
part of https://github.com/istio/istio/issues/41281

### CODE REVIEW

File: prow/config.yaml [hunk 1]
```diff
@@ -322,6 +322,19 @@ branch-protection:
...
+              protect: true
+              restrictions:
+                teams:
+                - zhlsunshine
+                - jacob-delgado
+            experimental-ambient:
+              protect: true
+              restrictions:
+                teams:
+                - stevenctl
```
Comment (zirain) [inline prow/config.yaml]: @howardjohn need a new team for this?

Comment (howardjohn) [reply prow/config.yaml]: Good question.. I was originally thinking we would just use the standard CODEOWNERS, but maybe we don't want to spam them all.

Although maybe the config.yaml shouldn't have this, and the forks should just modify CODEOWNERS as fit?

Comment (zirain) [reply prow/config.yaml]: for ambient branch, maybe the CODEOWNERS know less than people who really involved.

BTW change the CODESOWNERS file in target branch will also bring more stuff when you try merge it into master?

**Review by howardjohn — APPROVED** [review]

**Comment by zirain** [comment]
ping @istio/operations-maintainers

**Comment by howardjohn** [comment]
fyi: https://github.com/istio/istio/pull/41384

**Review by ericvn — APPROVED** [review]

**Review by linsun — APPROVED** [review]

**Comment by linsun** [comment]
@istio/operations-maintainers could someone approve this?

**Review by chaodaiG — APPROVED** [review]

**Pull request merged**